core/src/main/java/io/questdb/griffin/SqlCompilerImpl.java

```diff
@@ -375,6 +380,47 @@ public void compileBatch(
         }
     }
 
+    @Override
+    public void execute(final Operation op, SqlExecutionContext executionContext) throws SqlException {
```

Comment:
this is weird: you use SQL compiler to translate SQL text into CompiledQuery. That makes a lot of sense. But then when executing certain types of CompiledQuery we call the compiler again. That's very weird and I am sure future generations of QuestDB developers would appreciate at least some comments why is it like this.

Reply:
added doc here: https://github.com/questdb/questdb/pull/5159/commits/c44db28a716a873faaa4c76d7eaad2c55359aacc